### PR TITLE
Sounds

### PR DESCRIPTION
**Pull request by J3ns3its**

### CODE REVIEW

File: Assets/!/Scripts/PlayerAudio.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.IO;
+using utils;
+
+public class PlayerAudio : MonoBehaviour
+{
+    public AudioSource audioSource;
+    public List<Audio> audioPlayer = new List<Audio>();
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:10]: Use arrays since we wont update its sice on runtime

File: Assets/!/Scripts/PlayerAudio.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+        if (audioToPlay == null || !audioToPlay.HasValue)
+        {
+            Debug.LogWarning("Audio clip not found: " + clipName);
+            return;
+        }
+
+        _currentClip = audioToPlay.Value.clip;
+
+        if (_currentClip != null)
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:55]: Please add this check into the guard clause above

File: Assets/!/Scripts/PlayerAudio.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        {
+            audioSource.clip = _currentClip;
+            audioSource.volume = volume;
+            audioSource.Play();
+            audioSource.loop = false;
+
+        }
+    }
+}
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:64]: Missing line break

**Review by matias-gonz — CHANGES_REQUESTED** [review]

File: Assets/!/Scripts/PlayerAudio.cs (hunk 1 above)
Comment (bytsbyts) [reply Assets/!/Scripts/PlayerAudio.cs:10]: this change always leads problems of null check, is it possible we keep it as List

**Comment by bytsbyts** [comment]
Requested changes all solved.Please check

File: Assets/!/Scripts/PlayerAudio.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        // Add the check for audioToPlay.Value.clip being null into this guard clause
+        if (audioToPlay == null || audioToPlay.Value.clip == null)
+        {
+            UnityEngine.Debug.LogWarning("Audio clip not found: " + clipName);
+            return;
+        }
+
+        // Assign and check _currentClip in one step
+        _currentClip = audioToPlay.Value.clip;
+        if (_currentClip == null)
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:54]: This is already checked in the previous guard clause

**Review by matias-gonz — CHANGES_REQUESTED** [review]

File: Assets/!/Scripts/PlayerAudio.cs (hunk 4 above)
Comment (bytsbyts) [reply Assets/!/Scripts/PlayerAudio.cs]: Can you check now😊

File: Assets/!/Scripts/PlayerAudio.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+        if (audioToPlay == null || audioToPlay.Value.clip == null)
+        {
+            UnityEngine.Debug.LogWarning("Audio clip not found: " + clipName);
+            return;
+        }
+
+        _currentClip = audioToPlay.Value.clip;
+
+        audioSource.clip = _currentClip;
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:52]: nit: I see `_currentClip` is just used here, we can remove the attribute and just do this:
```suggestion
        audioSource.clip = audioToPlay.Value.clip;
```

**Review by matias-gonz — CHANGES_REQUESTED** [review]
Last nit! After that please solve conflicts and it is good 👍

File: Assets/!/Scripts/PlayerAudio.cs (hunk 5 above)
Comment (bytsbyts) [reply Assets/!/Scripts/PlayerAudio.cs]: It's changed now

File: Assets/!/Scripts/PlayerAudio.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
+using System.Diagnostics;
+using UnityEngine;
+using utils;
+
+public class PlayerAudio : MonoBehaviour
+{
+    public AudioSource audioSource;
+    public Audio[] audioPlayer;
+    private AudioClip _currentClip;
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:9]: We still need to remove the unused attribute

**Review by matias-gonz — CHANGES_REQUESTED** [review]

File: Assets/!/Scripts/PlayerAudio.cs (hunk 5 above)
Comment (matias-gonz) [reply Assets/!/Scripts/PlayerAudio.cs]: We still need to remove the unused attribute

File: Assets/!/Scripts/PlayerAudio.cs (hunk 6 above)
Comment (bytsbyts) [reply Assets/!/Scripts/PlayerAudio.cs]: It's done, sorry for didn't change completely

**Review by matias-gonz — COMMENTED** [review]
Solve conflicts and it is good to merge 🚀

**Pull request merged**